test/spec/features/preview-support/renderer/MarkerRenderer.js

```diff
@@ -1,141 +1,141 @@
-
-var CONNECTION_STYLE = {
-  fill: 'none',
-  stroke: 'fuchsia',
-  strokeWidth: 5
-};
-
-var MARKER_TYPES = [
-  'marker-start',
-  'marker-mid',
-  'marker-end'
-];
-
-/**
- * A renderer that can render markers.
- */
-export default function MarkerRenderer(canvas, eventBus, styles) {
-  DefaultRenderer.call(this, eventBus, styles, HIGH_PRIORITY);
-
-  this._canvas = canvas;
-
-  this._markers = {};
-}
-
-inherits(MarkerRenderer, DefaultRenderer);
-
-MarkerRenderer.$inject = [
-  'canvas',
-  'eventBus',
-  'styles'
-];
-
-MarkerRenderer.prototype.canRender = function() {
-  return true;
-};
-
-MarkerRenderer.prototype.drawConnection = function(parentGfx, connection) {
-  var line = createLine(connection.waypoints, CONNECTION_STYLE);
-
-  svgAppend(parentGfx, line);
-
-  var self = this;
-
-  MARKER_TYPES.forEach(function(markerType) {
-    if (hasMarker(connection, markerType)) {
-      self.addMarker(line, markerType);
-    }
-  });
-
-  return line;
-};
-
-MarkerRenderer.prototype.addMarker = function(gfx, markerType) {
-  var marker = this._markers[ markerType ],
-      defs;
-
-  if (!marker) {
-    marker = this._markers[ markerType ] = svgCreate('marker');
-
-    marker.id = markerType;
-
-    svgAttr(marker, {
-      refX: 5,
-      refY: 5,
-      viewBox: '0 0 10 10'
-    });
-
-    var circle = svgCreate('circle');
-
-    svgAttr(circle, {
-      cx: 5,
-      cy: 5,
-      fill: 'fuchsia',
-      r: 5
-    });
-
-    svgAppend(marker, circle);
-
-    defs = domQuery('defs', this._canvas._svg);
-
-    if (!defs) {
-      defs = svgCreate('defs');
-
-      svgAppend(this._canvas._svg, defs);
-    }
-
-    svgAppend(defs, marker);
-  }
-
-  var reference = idToReference(marker.id);
-
-  svgAttr(gfx, markerType, reference);
-};
-
-// helpers //////////
-
-/**
- * Get functional IRI reference for given ID of fragment within current document.
- *
- * @param {string} id
- *
- * @return {string}
- */
-function idToReference(id) {
-  return 'url(#' + id + ')';
-}
-
-/**
- * Check wether given connection has marker of given type.
- *
- * @param {Connection} connection
- * @param {string} markerType
- *
- * @return {boolean}
- */
-function hasMarker(connection, markerType) {
-  return connection.marker && connection.marker[ markerType.split('-').pop() ];
+import inherits from 'inherits-browser';
+
+import {
+  append as svgAppend,
+  attr as svgAttr,
+  create as svgCreate
+} from 'tiny-svg';
+
+import {
+  query as domQuery
+} from 'min-dom';
+
+import DefaultRenderer from 'lib/draw/DefaultRenderer';
+
+import {
+  createLine
+} from 'lib//util/RenderUtil';
+
+/**
+ * @typedef {import('../../model').Connection} Connection
+ */
+
+var HIGH_PRIORITY = 3000;
+
+var CONNECTION_STYLE = {
+  fill: 'none',
+  stroke: 'fuchsia',
+  strokeWidth: 5
+};
+
+var MARKER_TYPES = [
+  'marker-start',
+  'marker-mid',
+  'marker-end'
+];
+
+/**
+ * A renderer that can render markers.
+ */
+export default function MarkerRenderer(canvas, eventBus, styles) {
```

Comment:
What does/did this component do in the first place, and how did we change it? Maybe we could take the opportunity to document it appropriately?

Reply:
It's a simple renderer that renders markers for connections. We need it only for testing since the default renderer doesn't render markers.